src/exec.ml

```diff
@@ -180,63 +180,42 @@ let run_opam_package_deps ~repo packages =
   in
   run_and_log_l cmd
 
-let get_opam_field ~repo ~field package =
-  let field = field ^ ":" in
+let get_opam_fields_lines ~repo fields packages =
+  let fields = List.map (fun s -> s^":") fields in
+  let fields = "name"::fields in
+  let fields_str = String.concat ~sep:"," fields
+  in
+  let packages_str = List.map Types.Opam.string_of_package packages
+  in
   let cmd =
     let open Cmd in
     v "opam" % "show" %% switch_path repo % "--color=never" % "--normalise"
-    % "-f" % field % package
+    % "-f" % fields_str %% (of_list packages_str)
   in
-  run_and_log_s cmd
-  >>= fun r ->
-  match r with
-  | "" -> Ok OpamParserTypes.(List (("", 0, 0), []))
-  | r -> (
-    try Ok (OpamParser.value_from_string r "") with _ ->
-      Error (`Msg (Fmt.strf "parsing error for: '%s'" r)) )
-
-let get_opam_field_string_value ~repo ~field package =
-  get_opam_field ~repo ~field package
-  >>= fun v ->
-  let open OpamParserTypes in
-  match v with
-  | String (_, v) -> Ok v
-  | List (_, []) -> Ok ""
-  | _ ->
-      R.error_msg
-        (Fmt.strf
-           "Unable to parse opam string.\n\
-            Try `opam show --normalise -f %s: %s`"
-           field package)
-
-let get_opam_dev_repo ~repo package =
-  get_opam_field_string_value ~repo ~field:"dev-repo" package
-
-let get_opam_archive_url ~repo package =
-  get_opam_field_string_value ~repo ~field:"url.src" package
-  >>= function "" -> Ok None | uri -> Ok (Some uri)
-
-let get_opam_depends ~repo package =
-  get_opam_field ~repo ~field:"depends" package
-  >>= fun v ->
-  let open OpamParserTypes in
-  match v with
-  | List (_, vs) ->
-      let ss =
-        List.fold_left
-          (fun acc -> function String (_, v) -> v :: acc
-            | Option (_, String (_, v), _) -> v :: acc | _ -> acc )
-          [] vs
-      in
-      Logs.debug (fun l ->
-          l "Depends for %s: %s" package (String.concat ~sep:" " ss) ) ;
-      Ok ss
-  | _ ->
-      R.error_msg
-        (Fmt.strf
-           "Unable to parse opam depends for %s\n\
-            Try `opam show --normalise -f depends: %s` manually"
-           package package)
+  run_and_log_l cmd
+
+let get_opam_fields ~repo fields packages =
+  get_opam_fields_lines ~repo fields packages
+  >>= fun lines ->
+  let rec parse cur_name map = function
+  | [] -> Ok map
+  | line::next -> (
+    match String.cut ~sep:" " line with
+    | None -> parse cur_name map next
+    | Some ("name", value) -> parse (String.trim value) map next
+    | Some (field, value_str) ->
+      (match (String.trim value_str) with
+      | "" -> Ok OpamParserTypes.(List (("", 0, 0), []))
+      | r -> (
+        try Ok (OpamParser.value_from_string r "") with _ ->
```

Comment:
I think it would be better to only catch the relevant exception here, probably something like:
```suggestion
        try Ok (OpamParser.value_from_string r "") with OpamLexer.Error _ ->
```